### PR TITLE
adds document on Form styling

### LINKED ISSUE #518: Document how to add color and styling to Forms
https://nafundi.com/blog/posts/customize-odk-forms-with-fonts-colors-sizes-and-links/

Is this all still possible?

### LINKED ISSUE #627: Describe that form text can be styled using markdown
Some of this is described at http://xlsform.org/#styling

https://forum.opendatakit.org/t/how-can-i-add-styling-like-font-color-italics-etc-to-my-forms/12542 has a really nice form that shows all the styling options.

### PR DESCRIPTION
**Pull request by adammichaelwood**
closes #518 
closes #627 

#### What is included in this PR?

A document on styling form labels, hints, and choices using markdown or html.

### CODE REVIEW

File: odk1-src/form-styling.rst [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+  ### Header H3
+  
+  #### Header H4
+  
+  ##### Header H5
+  
+  ###### Header H6
+
+
+If a markdown header is used,
```
Comment (yanokwa) [inline odk1-src/form-styling.rst]: s/markdown/Markdown/g

File: odk1-src/form-styling.rst [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  :header: type, name, label, hint
+  
+  note, hyperlink, This label [contains a link](http://example.com)., This hint [contains a link](http://example.com).
+  
+.. _html-in-forms:
+  
+HTML
+-----
+
+Collect Forms support a subset of inline HTML elements.
```
[the remark this answers is not in the document]
Comment (adammichaelwood) [reply odk1-src/form-styling.rst]: >  so maybe we should say we use the fromHTML command and that's what drives this support.

I'm note sure if saying why something is arbitrary lessens the feelings of arbitrariness...

[the remark this answers is not in the document]
Comment (adammichaelwood) [reply odk1-src/form-styling.rst]: (I linked to a ref on the function. Maybe that helps?)

File: odk1-src/form-styling.rst [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+  
+  - `Styling prompts in XLSForm <http://xlsform.org/#styling>`_
+  - :download:`Sample XLSForm with Style </downloads/form-styling/style-example.xlsx>`
+
+.. _markdown-in-forms:
+
+Markdown
+---------
+
+:doc:`xlsform` supports limited used of `Markdown`_.
```
[the remark this answers is not in the document]
Comment (adammichaelwood) [reply odk1-src/form-styling.rst]: for got the reference
fixed

**Comment by lognaturel** [comment]
Looks good! There's some new info that would be great to get in there -- in Collect 1.15+, `#`, `\`, `_` and `*` can all be escaped by prepending a backslash.

`_hello_` -> _hello_
`\_hello\_` -> \_hello\_

File: odk1-src/form-styling.rst [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+  ### Header H3
+  
+  #### Header H4
+  
+  ##### Header H5
+  
+  ###### Header H6
+
+    
+.. image:: /img/form-styling/h1-label.*
```
Comment (yanokwa) [inline odk1-src/form-styling.rst]: It's fine to have the various headers, but the multiple screenshots don't add much. It's a lot of scrolling! Maybe we just have H1 and H6 and people can imagine what H2/3/4/5 look like?

Comment (yanokwa) [reply odk1-src/form-styling.rst]: Alternatively, H2-H5 could be hidden under show detail?

File: odk1-src/form-styling.rst [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+.. csv-table:: choices
+  :header: list_name, name, label
+  
+  pain, 1, 🙂
+  pain, 2, 😐
+  pain, 3, 🙁
+  pain, 4, 😦
+  pain, 5, 😧
+  pain, 6, 😩
+  pain, 7, 😱
```
Comment (yanokwa) [inline odk1-src/form-styling.rst]: Love this!

File: odk1-src/form-styling.rst (hunk 4 above)
Comment (adammichaelwood) [reply odk1-src/form-styling.rst]: Yeah. Something in an earlier comment from you (can't find now) made me think you were suggesting this. I originally wanted to put all six on one screen, but that doesn't actually work.
So --- if you think we don't really need to see all six sizes rendered, I'll pull them out.

Comment (yanokwa) [reply odk1-src/form-styling.rst]: One option is to put them all in a note, which gives you a sense of the size.

**Comment by adammichaelwood** [comment]
I really should add some alt text to these images, but other than that this is ready to go.
(And the alt text could be picked up in another PR if you want to get this in right away.)

**Comment by yanokwa** [comment]
This is looking good to me. Please add the alt text and we can merge then.

**Pull request merged**